src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -30,6 +31,7 @@
 @Inheritance(strategy = InheritanceType.SINGLE_TABLE)
 @DiscriminatorColumn(name = "discriminator", discriminatorType = DiscriminatorType.STRING)
 @DiscriminatorValue(value = "E")
+@DiscriminatorOptions(force = true)
```

Comment:
Why do we need this?

Reply:
This Annotation results in child entity repositories to not find an entity by id of a different child.
e.g. find a modeling exercise with id of a text exercise.
This case should return 404 instead of trying to cast it.